src/execution/mod.rs

```diff
@@ -49,26 +52,29 @@ use crate::recovery::{ProgressReader, StateCollector};
 use crate::recovery::{RecoveryConfig, StepId};
 use crate::recovery::{RecoveryStoreSummary, StateWriter};
 use crate::recovery::{StateBytes, StateKey};
-use crate::recovery::{StateReader, StateUpdate};
+use crate::recovery::StateReader;
```

Comment:
I don't think this is related to your PR but i'm curious about the way we're importing things. Why bundle some imports in `{}` and then do it again? Is this a formatting bug?